examples/model_compress/pruning/transformers/transformer_pruning.py

```diff
@@ -0,0 +1,596 @@
+
+    parser.add_argument("--model_name_or_path", type=str, required=True,
+                        help="Path to pretrained model or model identifier from huggingface.co/models.")
+    parser.add_argument("--task_name", type=str, default=None,
+                        help="The name of the glue task to train on.",
+                        choices=list(task_to_keys.keys()))
+    parser.add_argument("--output_dir", type=str, default=None,
+                        help="Where to store the final model.")
+    parser.add_argument('--usage', type=int, default=1,
+                        help='Select which config example to run')
+    parser.add_argument('--sparsity', type=float, required=True,
+                        help='Sparsity - proportion of heads to prune (should be between 0 and 1)')
+    parser.add_argument('--global_sort', action='store_true', default=False,
+                        help='Rank the heads globally and prune the heads with lowest scores. If set to False, the '
+                             'heads are only ranked within one layer')
+    parser.add_argument("--ranking_criterion", type=str, default='l1_weight',
+                        choices=["l1_weight", "l2_weight", "l1_activation", "l2_activation", "taylorfo"],
+                        help="Where to store the final model.")
+    parser.add_argument("--num_iterations", type=int, default=1,
+                        help="Number of pruning iterations (1 for one-shot pruning).")
+    parser.add_argument("--epochs_per_iteration", type=int, default=1,
+                        help="Epochs to finetune before the next pruning iteration "
+                             "(only effective if num_iterations > 1).")
+    parser.add_argument('--speed_up', action='store_true', default=False,
+                        help='Whether to speed-up the pruned model')
+
+    # parameters for model training; for running examples. no need to change them
+    parser.add_argument("--train_file", type=str, default=None,
+                        help="A csv or a json file containing the training data.")
+    parser.add_argument("--validation_file", type=str, default=None,
+                        help="A csv or a json file containing the validation data.")
+    parser.add_argument("--max_length", type=int, default=128,
+                        help=("The maximum total input sequence length after tokenization. Sequences longer than this "
+                              "will be truncated, sequences shorter will be padded if `--pad_to_max_lengh` is passed."))
+    parser.add_argument("--pad_to_max_length", action="store_true",
+                        help="If passed, pad all samples to `max_length`. Otherwise, dynamic padding is used.")
+    parser.add_argument("--use_slow_tokenizer", action="store_true",
+                        help="If passed, will use a slow tokenizer (not backed by the 🤗 Tokenizers library).",)
```

Comment:
🤗?

Reply:
This is from the original code. It seems that this emoji is called the "huggingface"

Reply:
we cannot copy code with Apache license though it is for user-friendly, because our repo is MIT license.

Reply:
we can try to provide instructions about how to get this transformer model and import it in our example code

Reply:
Fixed. Re-wrote the example from scratch.